### PR TITLE
change advanced relation store model

### LINKED ISSUE #9: [Improvement]: Default Value for Bool in Advanced ManyToMany Relation Column #10524
https://github.com/pimcore/pimcore/issues/10524

### PR DESCRIPTION
**Pull request by cameronfromtorq**
resolves #9

### CODE REVIEW

**Comment by cameronfromtorq** [comment]
@lukemacausland this is ready for you now

**Review by lukemacausland — APPROVED** [review]

File: bundles/AdminBundle/Resources/public/js/pimcore/object/tags/advancedManyToManyObjectRelation.js [hunk 1]
```diff
@@ -45,19 +45,29 @@ pimcore.object.tags.advancedManyToManyObjectRelation = Class.create(pimcore.obje
...
         for (i = 0; i < visibleFields.length; i++) {
-            fields.push(visibleFields[i]);
+            fields.push({name: visibleFields[i]});
         }
 
         for (i = 0; i < this.fieldConfig.columns.length; i++) {
-            fields.push(this.fieldConfig.columns[i].key);
+            var defaultValue = null;
+            switch(this.fieldConfig.columns[i].type.toLowerCase()){
+                case "bool":
```
Comment (evanjamesjackson) [inline bundles/AdminBundle/Resources/public/js/pimcore/object/tags/advancedManyToManyObjectRelation.js:62]: Any reason columnbool isn't included here like in the one below? I'm not even clear on what the difference is between those

Comment (cameronfromtorq) [reply bundles/AdminBundle/Resources/public/js/pimcore/object/tags/advancedManyToManyObjectRelation.js:62]: If I remember correctly, the Object version doesnt provide it as an option

Comment (evanjamesjackson) [reply bundles/AdminBundle/Resources/public/js/pimcore/object/tags/advancedManyToManyObjectRelation.js:62]: You're correct - I thought I saw it there in my test but nope

**Review by evanjamesjackson — APPROVED** [review]
Looks great

**Pull request closed without merging**